### PR TITLE
feat(crewai): Add SpanKind and Token Count attributes

### PR DESCRIPTION
**Pull request by harrisonchu**
resolves https://github.com/Arize-ai/phoenix/issues/4200

fix(crewai): add basic unit tests

### CODE REVIEW

File: python/examples/llama-index/backend/app/engine/loaders/__init__.py
Comment (fjcasti1) [inline python/examples/llama-index/backend/app/engine/loaders/__init__.py:1]: Undo these changes imo, they are unrelated to the goal of the PR

File: python/instrumentation/openinference-instrumentation-crewai/src/openinference/instrumentation/crewai/_wrappers.py [hunk 1]
```diff
@@ -205,15 +212,36 @@ def __call__(
                 ),
             )
             try:
-                response = wrapped(*args, **kwargs)
+                crew_output = wrapped(*args, **kwargs)
+                usage_metrics = instance.usage_metrics
```
Comment (fjcasti1) [inline python/instrumentation/openinference-instrumentation-crewai/src/openinference/instrumentation/crewai/_wrappers.py:216]: Seems like `crew=instance`, should we use one or the other?

File: python/instrumentation/openinference-instrumentation-crewai/src/openinference/instrumentation/crewai/_wrappers.py [hunk 2]
```diff
@@ -205,15 +212,36 @@ def __call__(
...
+                if isinstance(usage_metrics, dict):
+                    span.set_attribute(
+                        LLM_TOKEN_COUNT_PROMPT, int(usage_metrics.get("prompt_tokens", 0))
+                    )
+                    span.set_attribute(
+                        LLM_TOKEN_COUNT_COMPLETION, int(usage_metrics.get("completion_tokens", 0))
+                    )
+                    span.set_attribute(
+                        LLM_TOKEN_COUNT_TOTAL, int(usage_metrics.get("total_tokens", 0))
+                    )
```
Comment (fjcasti1) [inline python/instrumentation/openinference-instrumentation-crewai/src/openinference/instrumentation/crewai/_wrappers.py:226]: I don't think we should default to 0. If usage metrics does not have a given key, we should just not set the value, not assume 0. WDYT?

File: python/instrumentation/openinference-instrumentation-crewai/src/openinference/instrumentation/crewai/_wrappers.py [hunk 3]
```diff
@@ -205,15 +212,36 @@ def __call__(
...
                 span.set_status(trace_api.Status(trace_api.StatusCode.ERROR, str(exception)))
                 span.record_exception(exception)
                 raise
             span.set_status(trace_api.StatusCode.OK)
-            span.set_attribute(OUTPUT_VALUE, response)
+            if crew_output.to_dict():
+                span.set_attribute(OUTPUT_VALUE, json.dumps(crew_output.to_dict()))
+                span.set_attribute(OUTPUT_MIME_TYPE, "application/json")
+            else:
+                span.set_attribute(OUTPUT_VALUE, str(crew_output))
```
Comment (fjcasti1) [inline python/instrumentation/openinference-instrumentation-crewai/src/openinference/instrumentation/crewai/_wrappers.py:242]:
```suggestion
            if output_dict:=crew_output.to_dict():
                span.set_attribute(OUTPUT_VALUE, json.dumps(output_dict))
                span.set_attribute(OUTPUT_MIME_TYPE, "application/json")
            else:
                span.set_attribute(OUTPUT_VALUE, str(crew_output))
```

File: python/instrumentation/openinference-instrumentation-crewai/tests/openinference/instrumentation/crewai/test_instrumentor.py [hunk 4]
```diff
@@ -60,38 +144,75 @@ def test_crewai_instrumentation(
...
-    if use_context_attributes:
-        with using_attributes(
-            session_id=session_id,
-            user_id=user_id,
-            metadata=metadata,
-            tags=tags,
-            prompt_template=prompt_template,
-            prompt_template_version=prompt_template_version,
-            prompt_template_variables=prompt_template_variables,
+    with using_attributes(
```
Comment (fjcasti1) [inline python/instrumentation/openinference-instrumentation-crewai/tests/openinference/instrumentation/crewai/test_instrumentor.py:147]: Seems like these 2 tests could be combined into 1 no?

File: python/instrumentation/openinference-instrumentation-litellm/guides/client_demo.ipynb
Comment (fjcasti1) [inline python/instrumentation/openinference-instrumentation-litellm/guides/client_demo.ipynb:1]: Should this be included in this PR?

File: python/instrumentation/openinference-instrumentation-promptflow/examples/chat_flow_example_to_arize.ipynb
Comment (fjcasti1) [inline python/instrumentation/openinference-instrumentation-promptflow/examples/chat_flow_example_to_arize.ipynb:1]: Should this be included in this PR?

File: python/instrumentation/openinference-instrumentation-promptflow/examples/chat_flow_example_to_phoenix.ipynb
Comment (fjcasti1) [inline python/instrumentation/openinference-instrumentation-promptflow/examples/chat_flow_example_to_phoenix.ipynb:1]: Should this be included in this PR?

**Review by fjcasti1 — COMMENTED** [review]
Use footers to trigger release please release notes notes instead of making the title a "multi-title". See https://github.com/googleapis/release-please?tab=readme-ov-file#what-if-my-pr-contains-multiple-fixes-or-features

PS: would love a training session on vrcpy ;)

File: python/instrumentation/openinference-instrumentation-crewai/tests/openinference/instrumentation/crewai/test_instrumentor.py (hunk 4 above)
Comment (harrisonchu) [reply python/instrumentation/openinference-instrumentation-crewai/tests/openinference/instrumentation/crewai/test_instrumentor.py:147]: i thought breaking it out into two is actually more readable  because of the nesting with the `using_attributes` with the `use_casette` context manager.

File: python/instrumentation/openinference-instrumentation-litellm/guides/client_demo.ipynb
Comment (harrisonchu) [reply python/instrumentation/openinference-instrumentation-litellm/guides/client_demo.ipynb:1]: let me revert all of these, i think i accidentally ran ruff at the top level

File: python/instrumentation/openinference-instrumentation-crewai/src/openinference/instrumentation/crewai/_wrappers.py (hunk 1 above)
Comment (harrisonchu) [reply python/instrumentation/openinference-instrumentation-crewai/src/openinference/instrumentation/crewai/_wrappers.py:216]: good catch!

File: python/instrumentation/openinference-instrumentation-crewai/tests/openinference/instrumentation/crewai/test_instrumentor.py (hunk 4 above)
Comment (fjcasti1) [reply python/instrumentation/openinference-instrumentation-crewai/tests/openinference/instrumentation/crewai/test_instrumentor.py:147]: Ah, I missed the `use_casette`, my bad

File: python/instrumentation/openinference-instrumentation-crewai/src/openinference/instrumentation/crewai/_wrappers.py [hunk 5]
```diff
@@ -205,15 +212,33 @@ def __call__(
...
-                response = wrapped(*args, **kwargs)
+                crew_output = wrapped(*args, **kwargs)
+                usage_metrics = crew.usage_metrics
+                if isinstance(usage_metrics, dict):
+                    if "prompt_tokens" in usage_metrics:
+                        span.set_attribute(LLM_TOKEN_COUNT_PROMPT, int(usage_metrics["prompt_tokens"]))
+                    if "completion_tokens" in usage_metrics:
+                        span.set_attribute(LLM_TOKEN_COUNT_COMPLETION, int(usage_metrics["completion_tokens"]))
+                    if "total_tokens" in usage_metrics:
+                        span.set_attribute(LLM_TOKEN_COUNT_TOTAL, int(usage_metrics["total_tokens"]))
```
Comment (fjcasti1) [inline python/instrumentation/openinference-instrumentation-crewai/src/openinference/instrumentation/crewai/_wrappers.py:223]:
```suggestion
                    if prompt_tokens := usage_metrics.get("prompt_tokens"):
                        span.set_attribute(LLM_TOKEN_COUNT_PROMPT, int(prompt_tokens))
                    if  completion_tokens := usage_metrics.get("completion_tokens"):
                        span.set_attribute(LLM_TOKEN_COUNT_COMPLETION, int(completion_tokens))
                    if  total_tokens := usage_metrics.get("total_tokens"):
                        span.set_attribute(LLM_TOKEN_COUNT_TOTAL, int(total_tokens))
```

File: python/instrumentation/openinference-instrumentation-crewai/src/openinference/instrumentation/crewai/_wrappers.py [hunk 6]
```diff
@@ -205,15 +212,33 @@ def __call__(
...
+                else:
+                    # version 0.51 and onwards
+                    span.set_attribute(LLM_TOKEN_COUNT_PROMPT, usage_metrics.prompt_tokens)
+                    span.set_attribute(LLM_TOKEN_COUNT_COMPLETION, usage_metrics.completion_tokens)
+                    span.set_attribute(LLM_TOKEN_COUNT_TOTAL, usage_metrics.total_tokens)
```
Comment (fjcasti1) [inline python/instrumentation/openinference-instrumentation-crewai/src/openinference/instrumentation/crewai/_wrappers.py:228]: Are we absolutely certain these are set?

File: python/instrumentation/openinference-instrumentation-litellm/guides/client_demo.ipynb
Comment (fjcasti1) [inline python/instrumentation/openinference-instrumentation-litellm/guides/client_demo.ipynb:1]: Should this litellm notebook be in this PR?

**Review by fjcasti1 — COMMENTED** [review]
Reminder about the PR title (comment above)

File: python/instrumentation/openinference-instrumentation-crewai/src/openinference/instrumentation/crewai/_wrappers.py (hunk 6 above)
Comment (harrisonchu) [reply python/instrumentation/openinference-instrumentation-crewai/src/openinference/instrumentation/crewai/_wrappers.py:228]: yea in later versions it's set right before the end of the `kickoff` main function

**Review by fjcasti1 — APPROVED** [review]
